docs/part3-nextflow.md

```diff
@@ -0,0 +1,74 @@
+
+<!-- Talk about more nf things here -->
+
+### Step 1: Setting Up Your Workspace
+
+Before we start, we need make sure Nextflow is installed in your environment. This should already have been set up for you, but we can run commands like the following to see exactly what version of Nextflow we will be using:
+
+```bash
+nextflow -version
+```
+
+You should see the version info printed to the console.
+
+### Step 2: Understanding Your Nextflow Workflow
+
+Navigate to the `activity/nextflow/` directory where you will see a `main.nf` script. This script will define our workflow - let's take a closer look.
+
+Your workflow has several key components:
+
+a. Channel Creation
+
+In Nextflow, channels are used to transport data between processes. Think of them as pipelines through which your data flows. In this workflow, the `pics` channel is created from a file path specified by the parameter named `input`. This channel is responsible for making the images available to the first process (Classify) in the workflow.
+
+b. Parameters
+
+Parameters (`params`) allow for flexible, user-defined inputs into the workflow. They enable users to specify input directories (`params.input`), labels for classification (`params.prompts`), and output directories (params.outdir) at runtime. This flexibility is a stark contrast to our earlier Bash scripts, where such values would need to be hardcoded or passed in a less structured way.
+
+c. Processes
+`process Classify`: This process will execute the `classify` command we have been running in earlier steps manually, for each image based on the provided labels, and outputs the classification results to a text file (out.txt) alongside the original image.
+
+`process Collage`: After classification, this process takes groups of labeled images and creates a collage for each label. It uses the montage command from ImageMagick (a software suite for image manipulation). Recall, we were running two commands to achieve this earlier.
+
+`process CombineImages`: This final process gathers all the collages created by the earlier `Collage` process and combines them into one large image (collage.png). This demonstrates Nextflow's ability to aggregate outputs from previous steps and produce a consolidated result, a task that would be cumbersome and error-prone in bash, especially with varying numbers of intermediate files.
+
+### Step 3: Running the Workflow
+
+Now that you understand how the Workflow is structured, you can now start running some Nextflow magic!
+
+All of the data you will use as input for the workflow is provided in `activity/nextflow/data`. With Nextflow, initiating the workflow is as simple as running a command in your terminal, specifying your image folder and labels. Nextflow takes care of the rest, efficiently managing resources and ensuring the process is smooth.
+
+Run the following command on the command line:
```

Comment:
````suggestion
We have temporarily disabled some sections of the workflow, so that you can see how things change as we progressively add components. Right now the workflow section looks like:

```
workflow {
    pics = Channel.fromPath(params.input)

    Classify(pics, params.prompts)
    .view()
//    | map { label, pic -> [label.text, pic] }
//    | groupTuple
//    | Collage
//    | collect
//    | CombineImages
}
```

The `//` are 'comments', and will disable those lines for now. `.view()` lets us view the output of the Classify process in the meantime.

Run the following command on the command line:
````